src/dpl/src/CheckPlacement.cpp

```diff
@@ -139,21 +137,17 @@ void Opendp::checkPlacement(const bool verbose,
   }
 }
 
-void Opendp::processViolationsPtree(boost::property_tree::ptree& entry,
-                                    const std::vector<Cell*>& failures,
-                                    const string& violation_type) const
+void Opendp::saveViolations(const std::vector<Cell*>& failures,
+                            odb::dbMarkerCategory* category,
+                            const string& violation_type) const
 {
-  using boost::property_tree::ptree;
-  ptree violations;
-  const double dbUnits
-      = block_->getDataBase()->getTech()->getDbUnitsPerMicron();
   const Rect core = grid_->getCore();
   for (auto failure : failures) {
-    ptree violation, shapes, source, sources, shape;
-    double xMin = (failure->x_ + core.xMin()).v / dbUnits;
-    double yMin = (failure->y_ + core.yMin()).v / dbUnits;
-    double xMax = (failure->x_ + failure->width_ + core.xMin()).v / dbUnits;
-    double yMax = (failure->y_ + failure->height_ + core.yMin()).v / dbUnits;
+    odb::dbMarker* marker = odb::dbMarker::create(category);
```

Comment:
you need to check if this is a nullptr
If the category has reached it's limit, then this will return a `nullptr`